crates/transport_tx5/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+            K2Error::other_src("converting kitsune url to tx5 PeerUrl", e)
+        })
+    }
+}
+
+/// Tx5Transport configuration types.
+pub mod config {
+    /// Configuration parameters for [Tx5TransportFactory](super::Tx5TransportFactory).
+    #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Tx5TransportConfig {
+        /// The url of the sbd signal server. E.g. `wss://sbd.kitsu.ne`.
+        pub server_url: String,
+    }
+
+    impl Default for Tx5TransportConfig {
+        fn default() -> Self {
+            Self {
+                server_url: "<wss://your.sbd.url>".into(),
+            }
+        }
+    }
+
+    /// Module-level configuration for Tx5Transport.
+    #[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Tx5TransportModConfig {
+        /// Tx5Transport configuration.
+        pub tx5_transport: Tx5TransportConfig,
+    }
+}
+
+use config::*;
+
+/// Provides a Kitsune2 transport module based on the Tx5 crate.
+#[derive(Debug)]
+pub struct Tx5TransportFactory {}
+
+impl Tx5TransportFactory {
+    /// Construct a new Tx5TransportFactory.
+    pub fn create() -> DynTransportFactory {
+        let out: DynTransportFactory = Arc::new(Tx5TransportFactory {});
+        out
+    }
+}
+
+impl TransportFactory for Tx5TransportFactory {
+    fn default_config(&self, config: &mut Config) -> K2Result<()> {
+        config.set_module_config(&Tx5TransportModConfig::default())
+    }
+
+    fn create(
+        &self,
+        builder: Arc<builder::Builder>,
+        handler: DynTxHandler,
+    ) -> BoxFut<'static, K2Result<DynTransport>> {
+        Box::pin(async move {
+            let config: Tx5TransportModConfig =
+                builder.config.get_module_config()?;
+            let handler = TxImpHnd::new(handler);
+            let imp =
+                Tx5Transport::create(config.tx5_transport, handler.clone())
+                    .await?;
+            Ok(DefaultTransport::create(&handler, imp))
+        })
+    }
+}
+
+#[derive(Debug)]
+struct Tx5Transport {
+    ep: Arc<tx5::Endpoint>,
+    pre_task: tokio::task::JoinHandle<()>,
+    evt_task: tokio::task::JoinHandle<()>,
+}
+
+impl Drop for Tx5Transport {
+    fn drop(&mut self) {
+        self.pre_task.abort();
+        self.evt_task.abort();
+    }
+}
+
+type PreCheckResp = tokio::sync::oneshot::Sender<std::io::Result<()>>;
+type PreCheck = (tx5::PeerUrl, Vec<u8>, PreCheckResp);
+type PreCheckRecv = tokio::sync::mpsc::Receiver<PreCheck>;
+
+impl Tx5Transport {
+    pub async fn create(
+        config: Tx5TransportConfig,
+        handler: Arc<TxImpHnd>,
+    ) -> K2Result<DynTxImp> {
+        let (pre_send, pre_recv) = tokio::sync::mpsc::channel::<PreCheck>(1024);
+
+        let preflight_send_handler = handler.clone();
+        let tx5_config = Arc::new(tx5::Config {
+            signal_allow_plain_text: true,
```

Comment:
Is this just permitted for testing and it will always use TLS with `wss` SBD urls?

Reply:
As you say exactly. I can carry this forward into the config, since we'll eventually want that anyways.

Reply:
Done: https://github.com/holochain/kitsune2/pull/75/commits/e281c2bc6e603afff7f03884232bb495c94e6bb6